Content.Shared/Movement/Systems/SharedMoverController.cs

```diff
@@ -274,7 +279,16 @@ protected void HandleMobMovement(
             var walkSpeed = moveSpeedComponent?.CurrentWalkSpeed ?? MovementSpeedModifierComponent.DefaultBaseWalkSpeed;
             var sprintSpeed = moveSpeedComponent?.CurrentSprintSpeed ?? MovementSpeedModifierComponent.DefaultBaseSprintSpeed;
 
-            wishDir = AssertValidWish(mover, walkSpeed, sprintSpeed);
+            // ES START
+            var backwardsAngle = moveSpeedComponent?.BackwardsAngle ?? MovementSpeedModifierComponent.ESDefaultBackwardsAngle;
+            var rotNorm = worldRot.ToWorldVec().Normalized();
+            var velNorm = velocity.Normalized();
+            var cosAngle = Vector2.Dot(velNorm, rotNorm);
+            var threshold = new Angle(MathF.PI) - (backwardsAngle / 2);
+
+            forceWalk = cosAngle < Math.Cos(threshold.Theta);
+            wishDir = AssertValidWish(mover, walkSpeed, sprintSpeed, forceWalk);
+            // ES END
```

Comment:
Clarity of scope nitpick, using a block to put all these variables in a neat bundle so its clear when they're used.

```suggestion
            // ES START
            // Check if we're walking backwards within the given angle range, and if so force walking speed.
            {
                var backwardsAngle = moveSpeedComponent?.BackwardsAngle ?? MovementSpeedModifierComponent.ESDefaultBackwardsAngle;
                var rotNorm = worldRot.ToWorldVec().Normalized();
                var velNorm = velocity.Normalized();
                var cosAngle = Vector2.Dot(velNorm, rotNorm);
                var threshold = new Angle(MathF.PI) - (backwardsAngle / 2);
  
                forceWalk = cosAngle < Math.Cos(threshold.Theta);
                wishDir = AssertValidWish(mover, walkSpeed, sprintSpeed, forceWalk);
            }
            // ES END
```